File: artplayer-kotlin/src/main/java/org/salient/artplayer/AbsMediaPlayer.java
package org.salient.artplayer;

import android.view.Surface;

import java.util.Map;

/**
 *  Created by Mai on 2018/7/10
 * *
 *  Description: 视频播放器的抽象基类
 * *
 */
public abstract class AbsMediaPlayer {

    private Object dataSource;//正在播放的当前url或uri

    private Map<String, String> mHeaders;

    public abstract void start();

    public abstract void prepare();

    public abstract void pause();

    public abstract boolean isPlaying();

    public abstract void seekTo(long time);

    public abstract void release();

    public abstract long getCurrentPosition();

    public abstract long getDuration();

    public abstract void setSurface(Surface surface);

    public abstract void setVolume(float leftVolume, float rightVolume);

    public abstract void mute(boolean isMute);

    public abstract void setLooping(boolean isLoop);

    public Object getDataSource() {
        return dataSource;
    }

    /**
     * 设置播放地址
     *
     * @param dataSource 播放地址
     */
    public void setDataSource(Object dataSource) {
        this.dataSource = dataSource;
    }

    public Map<String, String> getHeaders() {
        return mHeaders;
    }

    /**
     * 设置请求头
     *
     * @param headers 播放地址请求头
     */
    public void setHeaders(Map<String, String> headers) {
        this.mHeaders = mHeaders;
    }
}

File: artplayer-kotlin/src/main/java/org/salient/artplayer/AbsMediaPlayer.kt
package org.salient.artplayer

import android.view.Surface

/**
 * Created by Mai on 2018/7/10
 * *
 * Description: 视频播放器的抽象基类
 * *
 */
abstract class AbsMediaPlayer {
    /**
     * 设置播放地址
     *
     * @param dataSource 播放地址
     */
    var dataSource: Any? = null //正在播放的当前url或uri: Any? = null
    var headers: Map<String, String>? = null

    abstract fun start()
    abstract fun prepare()
    abstract fun pause()
    abstract val isPlaying: Boolean
    abstract fun seekTo(time: Long)
    abstract fun release()
    abstract val currentPosition: Long
    abstract val duration: Long
    abstract fun setSurface(surface: Surface?)
    abstract fun setVolume(leftVolume: Float, rightVolume: Float)
    abstract fun mute(isMute: Boolean)
    abstract fun setLooping(isLoop: Boolean)

}